app/src/test/java/com/example/candyshop/CandyShopViewModelTest.kt

```diff
@@ -0,0 +1,101 @@
+package com.example.candyshop
+
+import app.cash.turbine.test
+import com.example.candyshop.api.CandyItemsRepository
+import com.example.candyshop.data.Resource
+import com.example.candyshop.data.model.Candy
+import com.example.candyshop.rules.TestDispatcherRule
+import com.example.candyshop.ui.CandyShopViewModel
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.impl.annotations.MockK
+import io.mockk.junit4.MockKRule
+import io.mockk.unmockkAll
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class CandyShopViewModelTest {
+
+    @get:Rule
+    val testDispatcher = TestDispatcherRule()
+
+    @get:Rule
+    val mockkRule = MockKRule(this)
+
+    @MockK
+    private lateinit var candyItemsRepository: CandyItemsRepository
+
+    private lateinit var candyShopViewModel: CandyShopViewModel
+
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this)
```

Comment:
check why we are applying here and in other tests MockkRule and try to remove what is not needed (duplicate work) from tests